x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/join/StubRelation.java

```diff
@@ -46,13 +48,28 @@ public StubRelation(Source source, List<Attribute> output) {
         this.output = output;
     }
 
+    /**
+     * Produces a StubRelation whose output is the merger of the {@code sourcePlan}'s output and {@code destinationPlan}'s referenced
+     * attributes.
+     */
+    public static StubRelation of(UnaryPlan destinationPlan, LogicalPlan sourcePlan) {
+        return new StubRelation(destinationPlan.source(), computeOutput(destinationPlan, sourcePlan));
+    }
+
+    /**
+     * Produces a new StubRelation whose output is the merger of this StubRelation's output and the {@code sourcePlan}'s output.
+     */
+    public StubRelation extendWith(LogicalPlan sourcePlan) {
+        return new StubRelation(source(), mergeOutputAttributes(sourcePlan.output(), output));
+    }
+
     /*
      * The output of a StubRelation must also include any synthetic attributes referenced by the source plan (union types is a great
      * example of those attributes that has some special treatment throughout the planning phases, especially in the EsRelation).
      */
-    public static List<Attribute> computeOutput(LogicalPlan source, LogicalPlan target) {
-        Set<Attribute> stubRelationOutput = new LinkedHashSet<>(target.output());
-        stubRelationOutput.addAll(source.references().stream().filter(Attribute::synthetic).toList());
+    private static List<Attribute> computeOutput(LogicalPlan destinationPlan, LogicalPlan sourcePlan) {
+        Set<Attribute> stubRelationOutput = new LinkedHashSet<>(sourcePlan.output());
+        stubRelationOutput.addAll(destinationPlan.references().stream().toList());
```

Comment:
Do we know what would fail if we didn't add the references here?

I'm asking because this looks like it's a fix for a broken intermediate state. It should only be needed if the LHS of the inline join fails to output some attributes that are needed for the aggregate node on the RHS of the inline join.

And if that's the case, then making the stub relation claim that it can provide these attributes is a little confusing.